zipline/pipeline/loaders/utils.py

```diff
@@ -0,0 +1,90 @@
+import numpy as np
+import pandas as pd
+from six import iteritems
+from six.moves import zip
+
+from zipline.utils.numpy_utils import np_NaT
+
+
+def next_date_frame(dates, announcement_dates):
+    """
+    Make a DataFrame representing simulated next earnings dates.
+
+    Parameters
+    ----------
+    dates : pd.DatetimeIndex.
+        The index of the returned DataFrame.
+    announcement_dates : dict[int -> pd.Series]
+        Dict mapping sids to an index of dates on which earnings were announced
+        for that sid.
+
+    Returns
+    -------
+    next_earnings: pd.DataFrame
+        A DataFrame representing, for each (label, date) pair, the first entry
+        in `earnings_calendars[label]` on or after `date`.  Entries falling
+        after the last date in a calendar will have `np_NaT` as the result in
+        the output.
+
+    See Also
+    --------
+    previous_earnings_date_frame
+    """
+    cols = {
+        equity: np.full_like(dates, np_NaT) for equity in announcement_dates
+    }
+    raw_dates = dates.values
+    for equity, earnings_dates in iteritems(announcement_dates):
```

Comment:
Are these meant to be more general than earnings_dates and announcement_dates now?